executor/builder.go

```diff
@@ -2177,6 +2185,22 @@ func markChildrenUsedCols(outputSchema *expression.Schema, childSchema ...*expre
 	return
 }
 
+// extractChildrenUsedColIdxs extract column indexes of child executors from children used columns mark
+func extractChildrenUsedColIdxs(childrenUsedColsMark [][]bool) [][]int {
+	childrenUsedColIdxs := make([][]int, len(childrenUsedColsMark))
+	for childIdx, childUsedColsMark := range childrenUsedColsMark {
+		if childrenUsedColIdxs[childIdx] == nil {
+			childrenUsedColIdxs[childIdx] = make([]int, 0, len(childUsedColsMark))
```

Comment:
1. what's this for ? If `childrenUsedColIdxs[childIdx] == nil` is true, `len(childUsedColsMark)` will be 0.
2. Can we append nil directly?
3. We can `continue` in this if-block?

Reply:
1. Actually, we don't need test `if childrenUsedColIdxs[childIdx] == nil`.
2. So on
3. So on

I think we could remove the `if childrenUsedColIdxs[childIdx] == nil` block totally?